src/mail/mail.service.ts

```diff
@@ -0,0 +1,100 @@
+import { Injectable } from "@nestjs/common";
+import axios, { AxiosResponse } from "axios";
+import * as ejs from "ejs";
+
+@Injectable()
+export class MailService {
+  private readonly mailgunAPIBaseURL = `${process.env.MAILGUN_API_URL}/${process.env.MAILGUN_EMAIL_DOMAIN}`;
+  private readonly emailFromString = `${process.env.MAILGUN_DEFAULT_NAME} <${process.env.MAILGUN_DEFAULT_EMAIL}>`;
+
+  // Private helper function to send a payload to mailgun API.
+  // Exists to reduce code redundancy.
+  private sendMailPayload(params: Record<string, any>): Promise<AxiosResponse> {
+    return axios.post(
+      `${this.mailgunAPIBaseURL}/messages`,
+      {} /* Mailgun needs the payload to be in params, leave body empty */,
+      {
+        auth: { username: "api", password: process.env.MAILGUN_API_KEY },
+        params
+      }
+    );
+  }
+
+  // Sends a simple email with a plain text body.
+  async sendTextEmail(
+    emailTo: string,
+    subject: string,
+    bodyText: string,
+    emailsCC?: [string],
+    emailsBCC?: [string]
+  ): Promise<[number, string]> {
```

Comment:
Return types should also be DTOs (interfaces) rather than "primitive types" like arrays.